mobile-app/app/screens/AppNavigator/screens/Dex/components/PoolPairCards/PoolPairCards.tsx

```diff
@@ -375,6 +185,245 @@ export function PoolPairCards ({
+      ? [mappedPair.tokenA.displaySymbol, mappedPair.tokenB.displaySymbol]
+      : yourPair.symbol.split('-')
+  const toRemove = new BigNumber(1)
+    .times((yourPair as WalletToken).amount)
+    .decimalPlaces(8, BigNumber.ROUND_DOWN)
+  const ratioToTotal = toRemove.div(mappedPair?.totalLiquidity?.token ?? 1)
+  const symbol = `${symbolA}-${symbolB}`
+
+  // assume defid will trim the dust values too
+  const tokenATotal = ratioToTotal
+    .times(mappedPair?.tokenA.reserve ?? 0)
+    .decimalPlaces(8, BigNumber.ROUND_DOWN)
+  const tokenBTotal = ratioToTotal
+    .times(mappedPair?.tokenB.reserve ?? 0)
+    .decimalPlaces(8, BigNumber.ROUND_DOWN)
+  const isExpanded = expandedCardIds.some((id) => id === yourPair.id)
+
+  const onCollapseToggle = (): void => {
+    if (isExpanded) {
+      setExpandedCardIds(
+        expandedCardIds.filter((id) => id !== yourPair.id))
+    } else {
+      setExpandedCardIds([...expandedCardIds, yourPair.id])
+    }
+  }
+
+  if (mappedPair === undefined) {
+    return <></>
+  }
+  return (
+    <ThemedView
+      dark={tailwind('bg-gray-800 border-gray-700')}
+      light={tailwind('bg-white border-gray-200')}
+      style={tailwind('p-4 mb-2 border rounded')}
+      testID={type === 'your' ? 'pool_pair_row_your' : 'pool_pair_row'}
+    >
+      <View
+        style={tailwind('flex flex-row justify-between w-full')}
+        testID={`pool_pair_row_${index}_${symbol}`}
+      >
+        <View style={tailwind('w-3/5 flex-row items-center')}>
+          <PoolPairTextSection
+            symbolA={symbolA}
+            symbolB={symbolB}
+          />
+          {type === 'available' && (
+            <TouchableOpacity
+              onPress={() => setFavouritePoolpair(yourPair.id)}
+              style={tailwind('p-1.5 flex-row items-center')}
+              testID={`favorite_${symbolA}-${symbolB}`}
+            >
+              <ThemedIcon
+                iconType='MaterialIcons'
+                name={isFavouritePair ? 'star' : 'star-outline'}
+                size={20}
+                light={tailwind(
+                  isFavouritePair ? 'text-warning-500' : 'text-gray-600'
+                )}
+                dark={tailwind(
+                  isFavouritePair ? 'text-darkwarning-500' : 'text-gray-300'
+                )}
+              />
+            </TouchableOpacity>
+          )}
+        </View>
+        {mappedPair?.apr?.total !== undefined && mappedPair?.apr?.total !== null && (
+          <APRSection
+            label={`${translate('screens/DexScreen', 'APR')}: `}
+            value={{
+              text: new BigNumber(
+                isNaN(mappedPair.apr.total) ? 0 : mappedPair.apr.total
+              )
+                .times(100)
+                .toFixed(2),
+              decimalScale: 2,
+              testID: `apr_${symbol}`,
+              suffix: '%'
+            }}
+          />
+        )}
+      </View>
+      {type === 'available'
+        ? (
+          <PriceRatesSection
```

Comment:
How long do we cache this? I tried swapping on playground. A few swaps in and prices are still not updated. 

Prices on the Pool Pair Cards (e.g)
1 DFI = 0.97 dBTC
1dBTC = 1.02 DFI

Reply:
We are not cashing token price, as we are using Global polling based on blockCount and network, so no need to fetch per page. on api end.

Reply:
Can you try on a local network? Start from scratch so prices for DFI = BTC is 1-1.

Do a few swaps. Check if the prices will change after some block updates

Reply:
Will be resolved on Whale side